http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala

```diff
@@ -332,34 +333,52 @@ object Boot extends IOApp {
       blockingEc <- ExecutionContexts.cachedThreadPool[F]
       semaphore <- Resource.eval(Semaphore[F](255L))
       blocker = Blocker.liftExecutionContext(blockingEc)
-      storage <- GoogleStorageService.resource[F](pathToCredentialJson, blocker, Some(semaphore))
-      retryPolicy = RetryPolicy[F](RetryPolicy.exponentialBackoff(30 seconds, 5))
-
-      sslContext <- Resource.eval(SslContextReader.getSSLContext())
-      httpClientWithCustomSSL <- blaze.BlazeClientBuilder[F](blockingEc, Some(sslContext)).resource
-      clientWithRetryWithCustomSSL = Retry(retryPolicy)(httpClientWithCustomSSL)
-      clientWithRetryAndLogging = Http4sLogger[F](logHeaders = true, logBody = false)(clientWithRetryWithCustomSSL)
 
       // This is for sending custom metrics to stackdriver. all custom metrics starts with `OpenCensus/leonardo/`.
       // Typing in `leonardo` in metrics explorer will show all leonardo custom metrics.
       // As best practice, we should have all related metrics under same prefix separated by `/`
       implicit0(openTelemetry: OpenTelemetryMetrics[F]) <- OpenTelemetryMetrics
         .resource[F](applicationConfig.leoServiceAccountJsonFile, applicationConfig.applicationName, blocker)
 
-      // Note the Sam client intentionally doesn't use clientWithRetryAndLogging because the logs are
-      // too verbose. We send OpenTelemetry metrics instead for instrumenting Sam calls.
-      samDao = HttpSamDAO[F](clientWithRetryWithCustomSSL, httpSamDaoConfig, blocker)
+      // Set up database reference
       concurrentDbAccessPermits <- Resource.eval(Semaphore[F](dbConcurrency))
       implicit0(dbRef: DbReference[F]) <- DbReference.init(liquibaseConfig, concurrentDbAccessPermits, blocker)
-      runtimeDnsCache = new RuntimeDnsCache(proxyConfig, dbRef, runtimeDnsCacheConfig, blocker)
-      kubernetesDnsCache = new KubernetesDnsCache(proxyConfig, dbRef, kubernetesDnsCacheConfig, blocker)
-      welderDao = new HttpWelderDAO[F](runtimeDnsCache, clientWithRetryAndLogging)
-      dockerDao = HttpDockerDAO[F](clientWithRetryAndLogging)
-      jupyterDao = new HttpJupyterDAO[F](runtimeDnsCache, clientWithRetryAndLogging)
-      rstudioDAO = new HttpRStudioDAO(runtimeDnsCache, clientWithRetryAndLogging)
+
+      // Set up DNS caches
+      hostToIpMapping <- Resource.eval(Ref.of(Map.empty[Host, IP]))
```

Comment:
just to understand the whole flow, `RuntimeDnsCache` and `KubernetesDnsCache` will update the `Ref` whenever we call `getHostStatus`....this does 2 things 1. populate cache (a mapping from basicaly runtime name -> `https://<uuid>.jupyter-$env.firecloud.org`) 2. return host name ( `https://<uuid>.jupyter-$env.firecloud.org`) 3. update `Ref`, which will be used by customDnsResolver to find the IP for `https://<uuid>.jupyter-$env.firecloud.org`

Reply:
Yes exactly :)

1. `ProxyService` wants to make a request
2. `ProxyService` calls `RuntimeDnsCache` to get the hostname based on the runtime project/name
3. `RuntimeDnsCache` updates the `Ref` with the host-IP mapping and returns the hostname
4. `ProxyService` makes the call via akka-http
5. akka-http calls custom resolver which looks up the IP from the hostname